### PR TITLE
🌱 Remove Go dependencies on k8s.io/kubectl, k8s.io/metrics, sigs.k8s.io/kustomize, ...

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Just noticed that we depend on kubectl & kustomize in CAPI (while investigating an issue with dependency bumps in CAPV).

This PR removes the dependency on kubectl and many other modules.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
Wow, that's a huge improvement 🎉 
/lgtm

File: go.mod [hunk 1]
```diff
@@ -47,15 +47,13 @@ require (
 	k8s.io/component-base v0.31.1
 	k8s.io/klog/v2 v2.130.1
 	k8s.io/kube-openapi v0.0.0-20240228011516-70dd3763d340
-	k8s.io/kubectl v0.31.1
```
Comment (sbueringer) [inline go.mod:50]: > [A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s)

**Review by fabriziopandini — COMMENTED** [review]
Nice catch!
/lgtm
/approve

**Pull request merged**